tests/unit/lib/DynamicArray_unittest.cc

```diff
@@ -0,0 +1,91 @@
+#include <gtest/gtest.h>
+#include <shogun/lib/DynamicArray.h>
+
+using namespace shogun;
+
+TEST(CDynamicArray, set_array)
+{
+	int32_t* array = (int32_t*)malloc(5 * sizeof(5));
+	CDynamicArray<int32_t> data;
+	CDynamicArray<int32_t> data1;
+	for (int32_t i = 0; i < 5; i++)
+	{
+		array[i] = i;
+	}
+
+	data.set_array(array, 5, false, true);
+	data1.set_array(array, 5, false, false);
+	EXPECT_EQ(data.get_num_elements(), 5);
+	for (index_t i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(data[i], i);
+		EXPECT_EQ(data1[i], i);
+	}
+
+	// inited by const array
+	const int32_t* c_array = array;
```

Comment:
i would definitely create a different test each of the different `set_array` methods in `CDynamicArray` not having it in one test